### PR TITLE
feat(new transform): Initial `concat` transform implementation

### LINKED ISSUE #1173: New `concat` transform
Hey,

we would love to see an option to concat multiple labels, to parse them in combination.

We have logs that contain only day, month and time in the logline. The date is in the filename. With concat the grok parse would be able to parse a date from multiple labels after concating.

### PR DESCRIPTION
**Pull request by foosinn**
Implements a new concat transform.

Still needs docs, i asume i can use #1216 as a hint for creating them?

fixes #1173

### CODE REVIEW

**Comment by binarylogic** [comment]
Thanks! This looks great. For docs, we just need to add a `./meta/transforms/concat.toml` file and then run `make generate`. That'll generate all necessary docs.

File: src/transforms/concat.rs [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+use serde::{Deserialize, Serialize};
+use string_cache::DefaultAtom as Atom;
+
+use lazy_static::lazy_static;
+
+#[derive(Deserialize, Serialize, Debug)]
+#[serde(deny_unknown_fields)]
+pub struct ConcatConfig {
+    pub target: Atom,
+    pub joiner: String,
```
Comment (Jeffail) [inline src/transforms/concat.rs]: Might be worth making this an `Option` and defaulting to a space when it's not specified.

**Comment by Jeffail** [comment]
My bad, I thought I'd save you trouble by deleting `vector.toml` directly but the commit isn't signed so it breaks our checks. It won't prevent us merging though once you've added docs.

**Comment by Jeffail** [comment]
Hey @foosinn, I've updated our documentation guidelines to hopefully make it less confusing: https://github.com/timberio/vector/blob/master/DOCUMENTING.md. If you haven't got time to add them yourself then let us know and I'll help get this finished up.

**Comment by binarylogic** [comment]
Once all tests pass we can merge.

**Comment by binarylogic** [comment]
It looks like tests are failing:

> error[E0050]: method `build` has 1 parameter but the declaration in trait `topology::config::TransformConfig::build` has 2
   --> src/transforms/concat.rs:26:14

Mind fixing this? Then we can merge.

File: src/transforms/concat.rs (hunk 1 above)
Comment (foosinn) [reply src/transforms/concat.rs]: Done :)

**Comment by foosinn** [comment]
Sorry for the long delay. Can you review the latest changes? Local tests are fine.

**Comment by binarylogic** [comment]
Nice, let's merge!

**Pull request merged**